stdlib/asm/crypto/fri.masm

```diff
@@ -0,0 +1,607 @@
+    dup.4
+    dup.1
+    sub
+    dup.4
+    ext2inv
+    swap
+    dup.2
+    mul
+    movdn.2
+    mul
+end
+
+#! Given ω, τ on stack top, this routine computes factor `ω/ (τ - ω)`
+#! of β, as written in formula https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#! only when index j is odd | j ∈ [0..64)
+#!
+#! Note, ω ∈ F_p | p = 2^64 - 2^32 + 1
+#!       τ ∈ Quadratic extension field over F_p
+#!
+#! Expected stack state looks like
+#!
+#! [ω, _, _, _, _, τ1, τ0, ...]
+#!
+#! Final stack state looks like
+#!
+#! [⍳1, ⍳0, _, _, _, _, τ1, τ0, ...] | ⍳ = ω/ (τ - ω)
+proc.compute_factor_of_beta_for_odd_index
+    dup.6
+    dup.1
+    sub
+    dup.6
+    ext2inv
+    swap
+    dup.2
+    mul
+    movdn.2
+    mul
+end
+
+#! Given ⍳, accumulator ν on stack top, this routine accumulates ⍳ into ν only when j is even | j ∈ [0..64)
+#! following formula described in https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#!
+#! Expected stack top
+#!
+#! [ω, ν1, ν0, τ1, τ0, mem_ptr_q - 1, ...]
+#!
+#! Final stack state
+#!
+#! [a3, a2, ν1', ν0', τ1, τ0, mem_ptr_q, ...]
+proc.accumulate_for_even_index
+    exec.compute_factor_of_beta_for_even_index
+
+    movup.6
+    add.1
+    movdn.6
+    push.0.0.0
+    dup.9
+    push.0
+    swap
+    mem_loadw
+
+    movup.5
+    movup.5
+    movup.5
+    movup.5
+    ext2mul
+    swap
+    movup.5
+    add
+    swap
+    movup.4
+    add
+    movdn.3
+    movdn.3
+end
+
+#! Given ⍳, accumulator ν on stack top, this routine accumulates ⍳ into ν only when j is odd | j ∈ [0..64)
+#! following formula described in https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#!
+#! Expected stack top
+#!
+#! [ω, a3, a2, ν1, ν0, τ1, τ0, mem_ptr_q, ...]
+#!
+#! Final stack state
+#!
+#! [ν1', ν0', τ1, τ0, mem_ptr_q, ...]
+proc.accumulate_for_odd_index
+    exec.compute_factor_of_beta_for_odd_index
+
+    ext2mul
+    swap
+    movup.3
+    add
+    swap
+    movup.2
+    add
+end
+
+#! Given τ and starting memory address to q ( FRI codeword with 64 evaluations ), this routine computes β
+#! using formula described in https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#!
+#! Expected stack state
+#!
+#! [τ1, τ0, mem_ptr_q, ...]
+#!
+#! Final stack state
+#!
+#! [β1, β0, τ1, τ0, mem_ptr_q, ...]
+proc.compute_beta
+    # subtract starting address by 1, because we first increment and then load
+    # onto stack
+    movup.2
+    sub.1
+    movdn.2
+
+    push.0.0 # accumulator for β
+
+    # for j = 0
+    push.1
+    exec.accumulate_for_even_index
+
+    # for j = 1
+    push.549755813888
+    exec.accumulate_for_odd_index
+
+    # for j = 2
+	push.70368744161280
+    exec.accumulate_for_even_index
+
+    # for j = 3
+    push.18446744069412487169
+    exec.accumulate_for_odd_index
+
+    # for j = 4
+    push.17293822564807737345
+    exec.accumulate_for_even_index
+
+    # for j = 5
+	push.8
```

Comment:
minor nit: for this and subsequent similar lines there is an extra indent in front of `push`.

Reply:
Interesting catch, my editor doesn't show this for some reason. Some sort of glitch in my tab/ space settings. Fixed.